app/assets/javascripts/discourse/app/lib/lightbox.js

```diff
@@ -1,96 +1,89 @@
-import I18n from "I18n";
-import loadScript from "discourse/lib/load-script";
-import { escapeExpression } from "discourse/lib/utilities";
-import { renderIcon } from "discourse-common/lib/icon-library";
-import { isAppWebview, postRNWebviewMessage } from "discourse/lib/utilities";
-import { spinnerHTML } from "discourse/helpers/loading-spinner";
+import { scheduleOnce } from "@ember/runloop";
+import loadScript, { loadCSS } from "discourse/lib/load-script";
+import { iconHTML } from "discourse-common/lib/icon-library";
 import User from "discourse/models/user";
 
+const docElement = document.querySelector("html");
+const isRTL = docElement.classList.contains("rtl");
+
+// which elements to show in the gallery
+const LIGHTBOX_SELECTOR = "*:not(.spoiler):not(.spoiled) a.lightbox";
+const NEXT_ICON = isRTL ? iconHTML("chevron-left") : iconHTML("chevron-right");
+const PREV_ICON = isRTL ? iconHTML("chevron-right") : iconHTML("chevron-left");
+const ANIMATION = "lg-fade";
+const INITIAL_ANIMATION = "";
+const HIDE_CONTROLS_DELAY = 3000; // hide controls after 3s
+
 export default function (elem, siteSettings) {
   if (!elem) {
     return;
   }
 
-  loadScript("/javascripts/jquery.magnific-popup.min.js").then(function () {
-    const lightboxes = elem.querySelectorAll(
-      "*:not(.spoiler):not(.spoiled) a.lightbox"
-    );
-    $(lightboxes).magnificPopup({
-      type: "image",
-      closeOnContentClick: false,
-      removalDelay: 300,
-      mainClass: "mfp-zoom-in",
-      tClose: I18n.t("lightbox.close"),
-      tLoading: spinnerHTML,
+  const hasLightboxes = elem.querySelectorAll(LIGHTBOX_SELECTOR).length;
+  const canDownload =
+    !siteSettings.prevent_anons_from_downloading_files || User.current();
 
-      gallery: {
-        enabled: true,
-        tPrev: I18n.t("lightbox.previous"),
-        tNext: I18n.t("lightbox.next"),
-        tCounter: I18n.t("lightbox.counter"),
-      },
+  if (hasLightboxes) {
+    const options = {
+      mode: ANIMATION,
+      startClass: INITIAL_ANIMATION, // prevents default zoom transition
+      selector: LIGHTBOX_SELECTOR,
+      hideBarsDelay: HIDE_CONTROLS_DELAY,
+      nextHtml: NEXT_ICON,
+      prevHtml: PREV_ICON,
+      download: canDownload,
+    };
 
-      ajax: {
-        tError: I18n.t("lightbox.content_load_error"),
-      },
+    // main lib
+    loadScript("/javascripts/lightgallery.min.js").then(() => {
+      // lib zoom module
+      loadScript("/javascripts/lg-zoom.min.js").then(() => {
+        // lib base css
+        loadCSS("/javascripts/lightgallery.min.css").then(() => {
+          scheduleOnce("afterRender", this, () => {
```

Comment:
scheduleOnce doesn't work with an anonymous function, it would be called each time, either make it a dedicated function, or just use schedule

Reply:
I've changed it to schedule :+1: